backend/src/instagram/instagram.service.ts

```diff
@@ -30,7 +30,6 @@ export class InstagramFeedService {
     async fetchAndSaveYouTubeFeed() {
         this.logger.log('Fetching Instagram feeds Started');
         await instagram.get('users/self/media/recent')
-            .then(this.filterAlreadyExistingFeeds)
```

Comment:
Have you tested this?. This will fail. `this.filterAlreadyExistingFeeds` was returning `data` elements from `feed` obejct, which is expected by `this.formatFeed` . This needs to be updated.